scripts/check_camera_sharpness.py

```diff
@@ -0,0 +1,214 @@
+
+    Args:
+        image: The image.
+        label: Text to be added.
+        position: Where to add the label (either "top" or "bottom").
+
+    Returns:
+        Image with added label.
+    """
+    label_height = 20
+
+    if position == "top":
+        offset = 0
+    elif position == "bottom":
+        offset = image.shape[0] - label_height
+    else:
+        ValueError(f"Invalid position '{position}'")
+
+    # darken label background
+    row_slice = slice(offset, (offset + label_height))
+    image[row_slice, :, :] = 0.3 * image[row_slice, :, :]
+
+    image = cv2.putText(
+        image,
+        label,
+        (5, offset + label_height - 5),
+        cv2.FONT_HERSHEY_SIMPLEX,
+        0.5,
+        (255, 255, 255),
+    )
+    return image
+
+
+def auto_canny_params(
+    image: np.ndarray, sigma: float = 0.33
+) -> typing.Tuple[float, float]:
+    median = np.median(image)
+    lower = max(0, (1.0 - sigma) * median)
+    upper = min(255, (1.0 + sigma) * median)
+
+    return lower, upper
+
+
+@dataclasses.dataclass
+class Params:
+    canny_threshold1: float
+    canny_threshold2: float
+    edge_mean_threshold: float
+
+    def set_canny_threshold1(self, value: float) -> None:
+        self.canny_threshold1 = value
+
+    def set_canny_threshold2(self, value: float) -> None:
+        self.canny_threshold2 = value
+
+    def set_edge_mean_threshold(self, value: float) -> None:
+        self.edge_mean_threshold = value
+
+
+def main():
+    argparser = argparse.ArgumentParser(description=__doc__)
+    argparser.add_argument(
+        "camera_id",
+        type=str,
+        help="""ID of the camera that is used.  If --pylon is set this refers
+            to the DeviceUserId, otherwise it is the index of the device.
+        """,
+    )
+    argparser.add_argument(
+        "--canny",
+        "-c",
+        type=float,
+        nargs=2,
+        default=(25.0, 250.0),
+        metavar=("THRESH1", "THRESH2"),
+        help="Threshold parameters for the Canny algorithm.  Default: %(default)s.",
+    )
+    argparser.add_argument(
+        "--threshold",
+        "-t",
+        type=float,
+        default=12.0,
+        help="""Threshold for the edge mean.  Images with an 'edge mean' below this
+            threshold are marked as too blurry.  Default: %(default)s.
+        """,
+    )
+    argparser.add_argument(
+        "--buffer-size",
+        type=int,
+        default=3,
+        help="Buffer size in seconds for the smoothed edge mean.",
+    )
+    args = argparser.parse_args()
+
+    camera_data = trifinger_cameras.camera.SingleProcessData()
+    camera_driver = trifinger_cameras.camera.PylonDriver(args.camera_id)
+
+    camera_backend = trifinger_cameras.camera.Backend(  # noqa
+        camera_driver, camera_data
+    )
+    camera_frontend = trifinger_cameras.camera.Frontend(camera_data)
+
+    params = Params(args.canny[0], args.canny[1], args.threshold)
+
+    mean_buffer = deque([], maxlen=10 * args.buffer_size)
```

Comment:
The factor 10 is necessary because of the 10 Hz frequency of the cameras right? I was confused by that at first so maybe a comment would help.

Reply:
Good point.  Yes, it refers to the camera frequency.  I'll use a variable for it.